website/orders/api/v1/views.py

```diff
@@ -0,0 +1,351 @@
+
+    def get_permission_object(self):
+        """Get the object to check permissions for."""
+        obj = self.kwargs.get("shift")
+        return obj.venue
+
+
+class OrderRetrieveUpdateDestroyAPIView(RetrieveUpdateDestroyAPIView):
+    """Order Retrieve Update Destroy API View."""
+
+    serializer_class = OrderSerializer
+    permission_required = "orders.can_order_in_venue"
+    permission_classes = [HasGuardianPermission]
+    queryset = Order.objects.all()
+
+    def update(self, request, *args, **kwargs):
+        """
+        Update an order.
+
+        This method needs the orders.can_manage_shift_in_venue permission
+        """
+        if self.request.user.has_perm("orders.can_manage_shift_in_venue", self.kwargs.get("shift").venue):
+            return super().update(request, *args, **kwargs)
+        else:
+            raise PermissionDenied
+
+    def destroy(self, request, *args, **kwargs):
+        """
+        Destroy an order.
+
+        This method needs the orders.can_manage_shift_in_venue permission
+        """
+        if self.request.user.has_perm("orders.can_manage_shift_in_venue", self.kwargs.get("shift").venue):
+            return super().destroy(request, *args, **kwargs)
+        else:
+            raise PermissionDenied
+
+    def get_queryset(self):
+        """Get the queryset."""
+        return self.queryset.filter(shift=self.kwargs.get("shift"))
+
+    def get_permission_object(self):
+        """Get the object to check permissions for."""
+        obj = self.kwargs.get("shift")
+        return obj.venue
+
+
+class ShiftListCreateAPIView(ListCreateAPIView):
+    """Shift List Create API View."""
+
+    serializer_class = ShiftSerializer
+    queryset = Shift.objects.all()
+
+    def get_queryset(self):
+        """Get the queryset."""
+        active = self.request.query_params.get("active", None)
+        if active is not None:
+            active = active == "true"
+            timezone = pytz.timezone(settings.TIME_ZONE)
+            current_time = timezone.localize(datetime.datetime.now())
+            if active:
+                return self.queryset.filter(start_date__lte=current_time, end_date__gte=current_time)
+            else:
+                return self.queryset.filter(Q(start_date__gte=current_time) | Q(end_date__lte=current_time))
+        else:
+            return self.queryset.all()
+
+    def create(self, request, *args, **kwargs):
+        """
+        Create a Shift.
+
+        API endpoint for creating a Shift.
+        Permission required: can_manage_shift_in_venue
+        :param request: the request
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return: super method or PermissionDenied when 'venue' POST data is None or when user does not have permission
+        in indicated venue
+        """
+        venue = request.data.get("venue")
+        if request.user.has_perm("orders.can_manage_shift_in_venue", venue):
```

Comment:
don't we have a nice permission class for this?

Reply:
Don't think so.

Reply:
Nope